### PR TITLE
Add "Keep disjoint features separate" to dissolve alg

### LINKED ISSUE #7405: Add "Keep disjoint features separate" option for dissolve algorithm (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#47837
Author: @nyalldawson
QGIS version: 3.26

**Add "Keep disjoint features separate" option for dissolve algorithm**

### PR Description:
If enabled, this option will cause disjoint features and parts to be exported as separate features (instead of as parts of a
single multipart feature).

Sponsored by City of Canning

To illustrate, given this set of input features:

image

by default the dissolve algorithm will export a single multipolygon feature:

image

With the new "keep disjoint features separate" option enabled we get 3 different output features instead:

image

When dissolving by class, the default behaviour gives 2 multi-part polygon features:

image

With the "keep disjoint features separate" option enabled we get 5 output features:

image

### Commits tagged with [need-docs] or [FEATURE]
"[feature][processing] Add \"Keep disjoint features separate\" option\nfor dissolve algorithm\n\nIf enabled, this option will cause disjoint features and parts to\nbe exported as separate features (instead of as parts of a\nsingle multipart feature)."

### PR DESCRIPTION
**Pull request by DelazJ**
Fixes #7405

@baswein I would appreciate if you could give this a look. Or if you have idea how we could better expose/combine the screenshots in the related issue report? Thanks.

### CODE REVIEW

File: docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -2160,16 +2160,20 @@ feature.
 All output geometries will be converted to multi geometries.
 In case the input is a polygon layer, common boundaries of adjacent
 polygons being dissolved will get erased.
+If enabled, the optional "Keep disjoint features separate" setting will
+cause disjoint features and parts to be exported as separate features
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Indeed. Done!

File: docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -2161,8 +2161,8 @@ All output geometries will be converted to multi geometries.
 In case the input is a polygon layer, common boundaries of adjacent
 polygons being dissolved will get erased.
 If enabled, the optional "Keep disjoint features separate" setting will
-cause disjoint features and parts to be exported as separate features
-(instead of parts of a single multipart feature).
+cause features and parts that do not overlap or touch to be exported as 
+separate feature (instead of parts of a single multipart feature).
```
Comment (baswein) [inline docs/user_manual/processing_algs/qgis/vectorgeometry.rst:2165]: I think features should be plural in "exported as separate features". Sorry I started a review session (not sure what that means). I thought maybe that would just let me edit the plural directly.

**Comment by DelazJ** [comment]
Using transparency to show the whole shape of the features is a great move I was too lazy to tweak. Thanks.
The image 2 is not that correct as you can't guarantee that it outputs a (nor b, BTW): it takes the first processed feature.
For the image 3, I'm not really fan of the "OR" text; the default output of the option is the middle image, the other one being a combination result (though worth showing, IMHO). maybe we could just go with more spacing between the outputs (and an appropriate caption). Do you think we could add a and b to the blue and purple features, leaving the yellow one unlabeled, to show predictable and unpredictable outputs or would that add confusion?
> I couldn't figure out how to add them as a pull request to your branch

It is tricky to replace images in someone's branch using only the web interface. But now that you have your patch-2 branch extracted from mine, it could be possible to upload your images to replace mines in your branch. And open a PR in my fork.
Otherwise, from command line it would be something like (from memory, sorry)
```
git remote add DelazJ https://github.com/DelazJ/QGIS-Documentation.git
git fetch DelazJ keepDisjoint
git checkout keepDisjoint
```
Do the changes, commit and pull request

**Comment by DelazJ** [comment]
> if you see any documentation that you think I could help with for the upcoming release let me know.

@baswein By documentation, you mean [PR](https://github.com/qgis/QGIS-Documentation/pulls) or [issues](https://github.com/qgis/QGIS-Documentation/milestone/16)? Help is welcome in everything. If you have particular topics you are interested in, let me know and I'll for sure find your targets 😄 
Since we have cooperated already on Processing matters, I'd be interested in your feedback on #7605, and #7608 (I couldn't run that one locally). There is also #7536 I'm working on: text is done (though I'll have open questions) but I didn't start working on images. So if you have any time to spend on that, much appreciated.
Thanks.

**Comment by baswein** [comment]
> The image 2 is not that correct as you can't guarantee that it outputs a (nor b, BTW)

I removed the A. I thought it might help show that it was one feature but I think it does make sense to only show the letters when the output is guaranteed.


> For the image 3, I'm not really fan of the "OR" text;

How about as 2 images? It follows the pattern we set elsewhere and takes up less of the precious horizontal space in the table:
image

**Comment by DelazJ** [comment]
> How about as 2 images?

Sold. 👍 
> but I think it does make sense to only show the letters when the output is guaranteed.

Then we should also add a/b to the purple/blue features in your new "dissolve all" picture.

**Comment by baswein** [comment]
> Then we should also add a/b to the purple/blue features in your new "dissolve all" picture.

How about this:
dissolve_disjoint_AorB

**Pull request merged**